### PR TITLE
MWPW-151392: Sorting button/menu issues on Mobile

### PR DESCRIPTION
**Pull request by mirafedas**
Changed the SWC version to 0.45.0, which includes the fix for the action menu component.

Resolves: [MWPW-151392](https://jira.corp.adobe.com/browse/MWPW-151392)

**Test URLs:**
- Before: https://main--  cc--adobecom.hlx.live/products/catalog/?martech=off&mep=off
- After: https://main--  cc--adobecom.hlx.page/products/catalog?milolibs=mwpw-151392-swc-version--milo--mirafedas&martech=off&mep=off
https://mwpw-151392-swc-version--milo--mirafedas.hlx.page/ - this is for passing the PSI check. As we don't have sidenav block in Milo, only in CC, it's impossible to create a test content page at the moment

### CODE REVIEW

File: libs/features/mas/package.json [hunk 1]
```diff
@@ -15,29 +15,29 @@
...
+    "@spectrum-web-components/progress-circle": "^0.45.0",
+    "@spectrum-web-components/radio": "^0.45.0",
+    "@spectrum-web-components/reactive-controllers": "^0.45.0",
+    "@spectrum-web-components/search": "^0.45.0",
+    "@spectrum-web-components/sidenav": "^0.45.0",
+    "@spectrum-web-components/split-button": "^0.45.0",
+    "@spectrum-web-components/tabs": "^0.45.0",
+    "@spectrum-web-components/theme": "^0.45.0",
+    "@spectrum-web-components/tooltip": "^0.45.0",
+    "@spectrum-web-components/tray": "^0.45.0",
```
Comment (yesil) [inline libs/features/mas/package.json:40]: this deps should not be needed.
I think they have been copied mistakenly during mas -> milo move.
Can you remove these and try again?

File: libs/deps/mas/plans-modal.js
Comment (yesil) [inline libs/deps/mas/plans-modal.js:1]: do we need this file in this PR?  if not needed, let's omit it.

File: libs/blocks/tabs/tabs.js [hunk 2]
```diff
@@ -68,11 +68,6 @@ function configTabs(config, rootElem) {
     const sel = rootElem.querySelector(id);
     if (sel) sel.click();
   }
-  const tabParam = new URLSearchParams(window.location.search).get('tab');
```
Comment (Axelcureno) [inline libs/blocks/tabs/tabs.js:71]: What these changes are needed for?

Comment (mirafedas) [reply libs/blocks/tabs/tabs.js]: It got there by mistake, I removed this change from PR, thank you for noticing this

File: libs/features/mas/commerce/stats.json
Comment (yesil) [inline libs/features/mas/commerce/stats.json:1]: stats.json files should not be commited.
can you remove it and add .gitignore entry?

**Review by yesil — APPROVED** [review]

**Review by 3ch023 — APPROVED** [review]

File: libs/features/mas/package.json (hunk 1 above)
Comment (mirafedas) [reply libs/features/mas/package.json]: Looks like they are needed, without them it's not possible to build web-components

**Review by rahulgupta999 — APPROVED** [review]

File: libs/deps/mas/merch-card-all.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
+\f\r]`,Y=/<(?:(!--|\/[^a-zA-Z])|(\/?[a-zA-Z][^>\s]*)|(\/?$))/g,$t=/-->/g,wt=/>/g,z=RegExp(`>|${He}(?:([^\\s"'>=/]+)(${He}*=${He}*(?:[^
```
[the remark this answers is not in the document]
Comment (Roycethan) [reply libs/deps/mas/merch-card-all.js:4]: @mirafedas Do you usually  dismiss or fix this security warning ?

[the remark this answers is not in the document]
Comment (mirafedas) [reply libs/deps/mas/merch-card-all.js:4]: @Roycethan this is the first time I see such a warning. I looked into it today, and it complains about the regExp that was introduced some time ago in a different PR, so I am not sure if I need to address it in this PR or not. Will discuss tomorrow with guys and let you know

**Comment by Roycethan** [comment]
@mirafedas 
I see the solution working for ios v 17 and higher ...but Sort button is not at all functional  in ios v16 and lower
checked in  iPhone 14 v16.5 and iPhone 13 v15.2
video attached: 
Screenshot 2024-08-05 at 4 38 22 PM
cc @yesil @3ch023

**Review by Axelcureno — APPROVED** [review]

File: libs/deps/mas/merch-card-all.js [hunk 4]
```diff
@@ -1,6 +1,8 @@
...
+\f\r]`,q=/<(?:(!--|\/[^a-zA-Z])|(\/?[a-zA-Z][^>\s]*)|(\/?$))/g,it=/-->/g,st=/>/g,C=RegExp(`>|${we}(?:([^\\s"'>=/]+)(${we}*=${we}*(?:[^
```
[the remark this answers is not in the document]
Comment (yesil) [reply libs/deps/mas/merch-card-all.js:2]: @mirafedas I propose to add `libs/deps/mas` to `paths-ignore` in `.github/codeql/codeql-config.yml`

We cannot really address those checks.

cc: @overmyheadandbody @mokimo any thoughts?

[the remark this answers is not in the document]
Comment (mokimo) [reply libs/deps/mas/merch-card-all.js:2]: Ideally even generated code doesn't have any security issues 😬 However it's a little harder to control albeit surely possible

[the remark this answers is not in the document]
Comment (yesil) [reply libs/deps/mas/merch-card-all.js:2]: I think we are not the other of the regex but it must be coming from lit.
if it is us, it should be highlighted in the src folder.

File: libs/features/mas/web-components/test/spectrum.js [hunk 5]
```diff
@@ -35,7 +35,3 @@ import '@spectrum-web-components/link/sp-link.js';
 import '@spectrum-web-components/tabs/sp-tabs.js';
 import '@spectrum-web-components/tabs/sp-tab.js';
 import '@spectrum-web-components/tabs/sp-tab-panel.js';
-
-import { Overlay } from '@spectrum-web-components/overlay/src/Overlay.js';
-
-window.__merch__spectrum_Overlay = Overlay;
```
Comment (yesil) [inline libs/features/mas/web-components/test/spectrum.js:41]: 👍

File: libs/features/mas/web-components/test/merch-twp-d2p.utils.js [hunk 6]
```diff
@@ -42,7 +43,9 @@ async function applyTemplate(templates, deeplink = true) {
         dialogBase.setAttribute('mode', 'fullscreenTakeover');
     }
 
-    const overlay = await window.__merch__spectrum_Overlay.open(dialogBase, {
+    const { base } = getConfig();
+    const overlay = await import(`${base}/features/spectrum-web-components/dist/overlay.js`);
```
Comment (yesil) [inline libs/features/mas/web-components/test/merch-twp-d2p.utils.js:47]: 👍

File: libs/features/spectrum-web-components/src/overlay.js [hunk 7]
```diff
@@ -1,9 +1,10 @@
 import '@spectrum-web-components/overlay/sp-overlay.js';
 import '@spectrum-web-components/overlay/overlay-trigger.js';
 import { Overlay } from '@spectrum-web-components/overlay/src/Overlay.js';
-export { SlottableRequestEvent } from '@spectrum-web-components/overlay/src/slottable-request-event.js';
-
+import { SlottableRequestEvent } from '@spectrum-web-components/overlay/src/slottable-request-event.js';
 
 // temporary workaround for dynamic imports
 // eslint-disable-next-line no-underscore-dangle
 window.__merch__spectrum_Overlay = Overlay;
```
Comment (yesil) [inline libs/features/spectrum-web-components/src/overlay.js:8]: can you remove this line, it is not needed anymore?

**Review by yesil — APPROVED** [review]

File: libs/features/spectrum-web-components/src/overlay.js (hunk 7 above)
Comment (mirafedas) [reply libs/features/spectrum-web-components/src/overlay.js:8]: It is still needed for the merch sidenav

**Comment by mirafedas** [comment]
@Roycethan could you please verify once more?

**Comment by Roycethan** [comment]
@mirafedas Verified and covered regression. One glitch i see hlx.live Vs hlx.page in Safari browser. where clicking on sorting button works on hlx.page (https://main--cc--adobecom.hlx.page/products/catalog?milolibs=mwpw-151392-swc-version--milo--mirafedas ) and doesn't on hlx.live ( https://main--cc--adobecom.hlx.live/products/catalog?milolibs=mwpw-151392-swc-version--milo--mirafedas ) (attached Mac Sonoma Safari browserstack video) - Not sure what could be the reason but before adding ready for stage , plz check and if you could resolve it then you can add the label on my behalf.


https://github.com/user-attachments/assets/3aceed3f-bb6d-459b-be87-83007df8153a

**Comment by mirafedas** [comment]
@Roycethan I added a fix for the .live in Safari, could you please double-check if everything works after this?

File: libs/features/spectrum-web-components/README.md [hunk 8]
```diff
@@ -4,6 +4,7 @@ This repository provides individual Spectrum Web Components for consumption in S
 This feature is mainly designed for consumption by the different merch blocks.
 
 Please know that SWC is heavy, impacts page performance and is not suitable for cases where a Consonant component is already available in Milo, e.g: button.
+Be aware that the polyfills from the `custom-element.js` must be loaded before SWC dependencies, otherwise if may cause issues (specifically in Safari).
```
Comment (yesil) [inline libs/features/spectrum-web-components/README.md:7]: `it may cause issues.`

File: libs/features/spectrum-web-components/README.md [hunk 9]
```diff
@@ -13,7 +14,7 @@ await Promise.all([
   import(`${base}/features/spectrum-web-components/dist/theme.js`),
   import(`${base}/features/spectrum-web-components/dist/button.js`),
   import(`${base}/features/spectrum-web-components/dist/search.js`),
-]);
+]);custom
```
Comment (yesil) [inline libs/features/spectrum-web-components/README.md:17]: typo?

**Review by yesil — COMMENTED** [review]
can you try `await import` instead of `loadScript` in `merch.js`?
Also, contrary to what I said in slack, I think we should `await polyfills()` in catalog before loading any swc code to make sure that custom-elements are loaded.

**Comment by mirafedas** [comment]
@yesil when I replace the `loadScript` with a dynamic import, it doesn't work. What is the reason why we want to replace it?

**Comment by yesil** [comment]
@mirafedas awaiting the loadScript doesn't seem to block.
I would like to load custom-elements before any other SWC dependency.
Just trying to be on the safe side.

**Comment by Roycethan** [comment]
> @Roycethan could you please verify once more?

@mirafedas  Works now on Safari

**Review by Roycethan — APPROVED** [review]

**Pull request merged**